grails-app/services/com/unifina/service/EthereumService.groovy

```diff
@@ -38,10 +38,11 @@ class EthereumService {
 
 	/** Checks if given user has registered the given Ethereum address in their Streamr profile */
 	boolean hasEthereumAddress(SecUser user, String ethereumAddress) {
-		// TODO: should we check that service == "ETHEREUM"?
-		IntegrationKey key = IntegrationKey.where {
-			(user == user) && (idInService == ethereumAddress)
-		}.find()
+		IntegrationKey key = IntegrationKey.createCriteria().get {
+			eq("user", user)
+			eq("idInService", ethereumAddress, [ignoreCase: true])
+			eq("service", IntegrationKey.Service.ETHEREUM)
```

Comment:
Is this ETHEREUM or ETHEREUM_ID?

Reply:
ETHEREUM_ID probably. I guess ETHEREUM is they keys that are added for ethereum module integration, and ETHEREUM_ID is for user id

Reply:
Both ETHEREUM and ETHEREUM_ID.

Old code worked because those are the only two types that exist. But good idea to add them in case there are others in the future.